randovania/games/fusion/exporter/patch_data_factory.py

```diff
@@ -53,17 +53,24 @@ def _create_pickup_dict(self, pickup_list: list[ExportedPickupDetails]) -> dict:
             if "source" in node.extra:
                 is_major = True
             if not pickup.is_for_remote_player and pickup.conditional_resources[0].resources:
-                resource = pickup.conditional_resources[0].resources[-1][0].extra["item"]
+                conditional_extras = pickup.conditional_resources[0].resources[-1][0].extra
+                resource = conditional_extras["item"]
+                jingle = conditional_extras.get("Jingle", "Minor")
+                message_id = conditional_extras.get("MessageID", None)
             else:
                 resource = "None"
 
             text = pickup.collection_text[0]
             sprite = pickup.model.name
 
-            custom_message = {}
+            item_message = {}
             # Special case where we ignore metroid dna right now, because that needs more patcher work.
             if text != self._placeholder_metroid_message:
-                custom_message = {"Languages": dict.fromkeys(self._lang_list, text), "Kind": "CustomMessage"}
+                item_message = {"Languages": dict.fromkeys(self._lang_list, text), "Kind": "CustomMessage"}
+
+            # Resources we want to specify a generic Message ID if we have not already provided custom messaging
+            if resource in ["InfantMetroid"] and not item_message and message_id is not None:
+                item_message = {"Kind": "MessageID", "MessageID": message_id}
```

Comment:
1) you should update the commments
2) why is the second if logic so complicated? you can just make it an `else`